### PR TITLE
BROKERS: Lab Broker Get and Post Labs

### PR DESCRIPTION
**Pull request by chaehomie**
------------------------------------
DMX.Sdk core functions status:
➡️ RetrieveAllLabsAsync
🔜 UploadArtifactAsync
🔜 SendCommandAsync
------------------------------------

### CODE REVIEW

**Review by geetika-jain — APPROVED** [review]

File: DMX.Sdk/Brokers/Loggings/LoggingBroker.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            this.logger = logger;
+
+        public void LogInformation(string message) =>
+            this.logger.LogInformation(message);
+
+        public void LogTrace(string message) =>
+            this.logger.LogTrace(message);
+
+        public void LogDebug(string message) =>
+            this.logger.LogDebug(message);
```
Comment (hassanhabib) [inline DMX.Sdk/Brokers/Loggings/LoggingBroker.cs:24]: `LoggingBroker` should be it's own PR.

File: DMX.Sdk/Brokers/DmxApis/DmxApiBroker.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+            string apiBaseUrl = localConfigurations.ApiConfiguration.Url;
+            this.httpClient.BaseAddress = new Uri(apiBaseUrl);
+
+            return new RESTFulApiFactoryClient(this.httpClient);
+        }
+
+        private string[] GetScopesFromConfiguration(string scopeCategory)
+        {
+            LocalConfiguration localConfiguration =
```
Comment (hassanhabib) [inline DMX.Sdk/Brokers/DmxApis/DmxApiBroker.cs:60]: In SDKs - pulling in configurations might be a bit different. Have we tested getting these configurations from an `appSettings.json` file? SDKs are slightly different in the way they operate than an ASP.NET Core application

File: DMX.Sdk/Brokers/DmxApis/DmxApiBroker.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.Identity.Web;
+using RESTFulSense.Clients;
+
+namespace DMX.Sdk.Brokers.DmxApis
+{
+    public partial class DmxApiBroker : IDmxApiBroker
+    {
+        private HttpClient httpClient;
+        private readonly ITokenAcquisition tokenAcquisition;
+        private readonly IConfiguration conifguration;
```
Comment (hassanhabib) [inline DMX.Sdk/Brokers/DmxApis/DmxApiBroker.cs:17]: In the case of this SDK, the token will be acquired before hand by the consumer of the SDK. And they will need to provide that token on initialization of the library so the library can authenticate.

File: DMX.Sdk/Brokers/DmxApis/DmxApiBroker.Labs.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public partial class DmxApiBroker
+    {
+        private const string LabsRelativeUrl = "api/labs";
+
+        public async ValueTask<List<Lab>> GetAllLabsAsync()
+        {
+            await GetAccessTokenForScope("GetAllLabs");
+
+            return await GetAsync<List<Lab>>(LabsRelativeUrl);
+        }
```
Comment (hassanhabib) [inline DMX.Sdk/Brokers/DmxApis/DmxApiBroker.Labs.cs:18]: Each one of these functions `GetAllLabsAsync` and `PostLabAsync` requires their own PRs

**Comment by hassanhabib** [comment]
Shutting down this PR as we don't need it anymore.
